api/domain_handlers.go

```diff
@@ -381,3 +382,70 @@ func (api API) Remove(w http.ResponseWriter, r *http.Request) {
 
 	writeJSONOrBust(w, issues)
 }
+
+// RemoveIneligibleDomains runs eligibility checks on domains present in the
+// database and change the status to PendingAutomatedRemoval if the domain
+// does not follow the requirements for more than 2 crawls
+
+// Example: GET /removeineligibledomains?
+func (api API) RemoveIneligibleDomains(w http.ResponseWriter, r *http.Request) {
+
+	var ineligibleDomains []database.IneligibleDomainState
+	var deleteEligibleDomains []string
+
+	domains, err := api.database.AllDomainStates()
+	if err != nil {
+		msg := fmt.Sprintf("Internal error: could not retrieve domains. (%s)\n", err)
+		http.Error(w, msg, http.StatusInternalServerError)
+		return
+	}
+	// Store ineligible domains in slice
+	for _, d := range domains {
+		if d.Policy == preloadlist.Bulk18Weeks || d.Policy == preloadlist.Bulk1Year {
+
+			_, issues := api.hstspreload.EligibleDomain(d.Name, d.Policy)
+			state, err := api.database.GetIneligibleDomainStates([]string{d.Name})
```

Comment:
Multiple repeated calls to `GetIneligibleDomainStates` isn't a good access pattern - it would be better to make one call with all domains we care about. One way to do this would be to filter the domains slice to have only the ones with policies we care about, then make this call, and then iterate over the filtered domain slice. (For ease of access, it might be convenient to make a map from domain name to IneligibleDomainState.)

Another potentially better way would be to call `GetAllIneligibleDomainStates`. The thought here is that if there's an IneligibleDomainState for a domain that's no longer on the preload list, then we should remove it. This makes logic a little more complicated in the body of this function to identify such domains. One potential way to track that is if you have a map of domains to IneligibleDomainState, you could delete from that map when you process domains, and the ones remaining in the map are the ones that need to be removed.